petastorm/reader.py

```diff
@@ -241,6 +346,18 @@ def __init__(self, pyarrow_filesystem, dataset_path, schema_fields=None,
             to the main data store is either slow or expensive and the local machine has large enough storage
             to store entire dataset (or a partition of a dataset if shards are used).
             By default, use the :class:`.NullCache` implementation.
+
+        :param infer_schema: Whether to infer the unischema object from the parquet schema.
+            Only works for schemas containing certain scalar type. This option allows getting around explicitly
+            generating petastorm metadata using :func:`petastorm.etl.dataset_metadata.materialize_dataset` or
+            petastorm-generate-metadata.py. Should always be True for non-Petastorm generated datasets.
+
+        :param worker_class: This is the class that will be instantiated on a different thread/process. It's
+            responsibility is to load and filter the data.
+        :param results_queue_reader: An instance of an object that will be reading data produced by worker class.
```

Comment:
Since this is tightly coupled with the worker class does it make sense to have the results_queue_reader as a separate parameter?

Reply:
Fixed.